mod/groups/actions/groups/edit.php

```diff
@@ -19,6 +19,14 @@ function profile_array_decoder(&$v) {
 foreach (elgg_get_config('group') as $shortname => $valuetype) {
 	$input[$shortname] = get_input($shortname);
 
+	if ($input[$shortname] === '') {
+		// The group profile displays all profile fields that have a value.
+		// We don't want to display fields with empty string value, so we
+		// prevent it getting saved by setting the value to null.
+		$input[$shortname] = null;
```

Comment:
I think we just handle this where we copy $input to metadata. If it's empty string, deleteMetadata() instead.